server/broadcast.go

```diff
@@ -978,6 +1005,11 @@ func nonRetryableErrMapInit() map[string]bool {
 
 var NonRetryableErrMap = nonRetryableErrMapInit()
 
-func isNonRetryableError(e error) bool {
-	return NonRetryableErrMap[e.Error()]
+func isNonRetryableError(err error) bool {
+	for e := err; e != nil; e = errors.Unwrap(e) {
```

Comment:
Neat - so this keeps unwrapping an error and if at any point the error is non-retryable we return true. Curious if this was needed specifically to handle a case where we actually needed to detect a non-retryable error that was wrapped?

Reply:
Yeah! Basically the same logic as `errors.Is` and `errors.As` from the stdlib :)

It was exactly fixing a test that broke, because of the logic change for retrying the transcode attempts. Previously, if we got a non-retriable error on the last attempt, we would just return immediately from inside the `for` loop. Now, we always finish the `for` loop so the transcode event can be sent. That cause the non-retriable events to be wrapped on a `"hit max transcode attempts: %w"` error and then this function stopped detecting them correctly.